[WIP] Restructure and rewrite of structure and function predictions (#1003)
greenelab/deep-review: content/04.study.md

Earlier attampts of applying deep learning to secondary structure prediction [@doi:10.1371/journal.pone.0032235; @doi:10.1109/TCBB.2014.2343960; @doi:10.1038/srep11476] didn't achieve significant improvenets over the widly used and relatively simple PSIPRED [@doi:10.1006/jmbi.1999.3091].
In 2014, Zhou and Troyanskaya demonstrated that they could improve Q8 accuracy by using a deep supervised and convolutional generative stochastic network [@arxiv:1403.1347].
In 2016 Wang et al. developed a DeepCNF model that improved Q3 and Q8 accuracy as well as prediction of solvent accessibility and disorder regions [@doi:10.1038/srep18962; @doi:10.1007/978-3-319-46227-1_1].
Following these successes, recently proposed methods achieve significantly better results than PSIPRED using Deep Learning architectures, with significantly better DL methods today being NetSurfP-2.0 [@doi:10.1002/prot.25674] and Porter 5 [@doi:10.1038/s41598-019-48786-x].

The improvement obtained by these methods may be mainly due to the ability of convolutional neural fields to capture long-range information.
Top methods still heavily rely on the creation of profiles from Multiple Sequence Alignments (MSAs).

> Models relying on LMs have yet to reach the accuracy of evolutionary based methods, but are able to deliver results for proteins for which MSAs can't be computed and in general execute at a fracion of the time with respect to evolutionary based approaches [@doi:10.1186/s12859-019-3220-8].

Reviewer: ```suggestion
Models relying on LMs have yet to reach the accuracy of evolutionary based methods even for proteins for which good MSAs can't be built, but these LM methods in general execute at a fraction of the time with respect to evolutionary based approaches [@doi:10.1186/s12859-019-3220-8].
```

Author: @j3xugit

I think your addition has not yet been experimentally validated (`even for proteins for which good MSAs cannot be built`), and is, in large part, a repetition of the first part of the sentence (`Models relying on LMs have yet to reach the accuracy of evolutionary based methods`), which is the best (to my knowledge) one can say about the comparison in accuracy between evo vs. LMs.

My original intent (`but are able to deliver results for proteins for which MSAs can't be computed`) was meant in the sense that evo models (at least the ones I know) actually fail to produce results if MSAs can't be built.

If we can't agree, I'm happy to leave out this second part of the sentence. I would also rewrite the second part into its own sentence and make it clearer, here my suggestion:

```suggestion
Models relying on LMs have yet to reach the accuracy of evolutionary based methods.
On the upside, LMs require a fraction of the resources for inference compared to evolutionary based approaches [@doi:10.1186/s12859-019-3220-8].
```

Reviewer: It is incorrect to say that evo models fail to produce results if MSAs can't be built.  In fact my deep models (and other similar ones) works well on some proteins for which MSAs cannot be built. Baker group has also shown that the deep models developed by his group (which is similar to mine) works well on a good portion of human-designed proteins although MSAs cannot be built for them.

Reviewer 2: Are there specific references we should have in mind to support the discussion in these sentences?  That might help make sure all of us are looking at the same models and results.

Author: Right, I might have been too black and white here :) sorry about that!

My belief is that if a model bases predictions on only PSSMs as inputs (or evo couplings), but for a certain dark protein there simply isn't an MSA to start with, those predictions can't be trusted (I picture this as having an input matrix with all zeroes, just to give an idea). It's more of a "conceptual" point than reality ot things.
Models out there today do all sorts of things, including combining sequence-based features (often learned, e.g. via CNNs) with MSA extracted features, so mine would ba simplification.

I'm happy either way :) The more fundamental point of this sentence, for me, is that we still don't have a clear understanding on how well LMs work on those proteins for which also MSA-based methods perform arguably well, and how much we are buying ourselves coverage by using these models instead (or combinations of the two). But results for that will be out in Dec with CASP, I feel.

Reviewer 2: @sacdallago can you please edit these sentences to take into account @j3xugit's comment and the trRosetta results for de novo proteins?  Or @j3xugit could suggest new text for @sacdallago to review.

https://www.pnas.org/content/117/3/1496 (`doi:10.1073/pnas.1914677117`) could also be a relevant reference to add.

Author: If it's ok, I'll edit this the week of Aug 31st (I'm currently on holidays and it's hard to get into the writing headspace -- especially without a laptop & context overview)

Reviewer 2: Sending a brief reminder about these edits @sacdallago.  It shouldn't require anything too extensive.

Author: @agitter 

1. Thanks very much for the reminder -- unfortunately I had to attach 1 week leave of absence to my holidays due to some unfortunate family circumstances. AKA: sorry for the delay
2. I don't quite know how to integrate the TrRosetta paper in exactly this section. TrRosetta doesn't use language models to extract additional features, it's still relies exclusively on MSAs (from my understanding). I did add it to an earlier sentence where it appeared to make more sense.
3. I updated some other sections, which in the meantime have seen some new pre-prints and work.

I'll update the PR in about 10 min with the latest changes

#### Contact and distance maps
Contact prediction is much more challenging than secondary structure prediction, but provides much more information for 3D structure prediction.
Contact maps can broadly speaking be seen as distance maps with a specific distance cutoff applied.
In recent years, the accuracy of contact prediction has greatly improved [@doi:10.1371/journal.pcbi.1005324; @doi:10.1093/bioinformatics/btu791; @doi:10.1073/pnas.0805923106; @doi:10.1371/journal.pone.0028766].

> Protein contact prediction and contact-assisted folding (i.e. folding proteins using predicted contacts as restraints) represents a promising new direction for *ab initio* folding of proteins without good templates in PDB.

Reviewer: ```suggestion
Protein contact prediction and contact-assisted folding (i.e. folding proteins using predicted contacts as restraints) has proven to be a successful direction for *ab initio* folding of proteins without good templates in PDB.
```

However, blindly tested in the well-known CASP competitions, these methods did not show any advantage over MetaPSICOV [@doi:10.1093/bioinformatics/btu791].

RaptorX-Contact [@doi:10.1371/journal.pcbi.1005324] significantly improved contact prediction over MetaPSICOV and pure co-evolution methods, especially for proteins without many sequence homologs.
It employed a network architecture formed by one 1D residual neural network and one 2D residual neural network. Blindly tested in the latest Critical Assesment of Structure Prediction (CASP) competition (i.e. CASP12 [@url:http://www.predictioncenter.org/casp12/rrc_avrg_results.cgi]), RaptorX-Contact ranked first in F₁ score on free-modeling targets as well as the whole set of targets. In CAMEO (which can be interpreted as a fully-automated CASP) [@url:https://www.cameo3d.org], its predicted contacts were also able to fold proteins with a novel fold and only 65--330 sequence homologs.
This technique also worked well on membrane proteins even when trained on non-membrane proteins [@arxiv:1704.07207].

> RaptorX-Contact performed better mainly due to introduction of residual neural networks and exploitation of contact occurrence patterns by simultaneously predicting all the contacts in a single protein.

Reviewer: ```suggestion
RaptorX-Contact performed better mainly due to introduction of very deep convolutional residual neural networks and exploitation of contact occurrence patterns by simultaneously predicting all the contacts in a single protein.
```

> Most methods until late 2018 relied on just one distance threshold, simply defined by the ideal contact distance, to produce contact maps for 3D structure prediction.

Reviewer: ```suggestion
Right after its success on contact prediction, deep convolutional residual network has been used to predict inter-atom distance prediction and demonstrated its effectiveness in template-based protein modeling [@doi:10.1093/bioinformatics/bty278].
```

> Recent seminal work presented at the 13th edition of CASP combined multiple distance cutoffs together with torsion prediction and a new folding pipeline to achieve state of the art results [@doi:10.1038/s41586-019-1923-7].

Reviewer: ```suggestion
Recent seminal work presented at the 13th edition of CASP combined multiple distance cutoffs together with torsion prediction and a new folding pipeline to achieve state of the art results [@doi:10.1073/pnas.1821309116; @doi:10.1038/s41586-019-1923-7].
```
